cmd/api/src/daemons/datapipe/ingest.go

```diff
@@ -44,21 +45,46 @@ func ReadFileForIngest(batch graph.Batch, reader io.ReadSeeker) error {
 	}
 }
 
-func IngestBasicData(batch graph.Batch, converted ConvertedData) {
-	IngestNodes(batch, ad.Entity, converted.NodeProps)
-	IngestRelationships(batch, ad.Entity, converted.RelProps)
+func IngestBasicData(batch graph.Batch, converted ConvertedData) error {
+	errs := util.NewErrorCollector()
+
+	if err := IngestNodes(batch, ad.Entity, converted.NodeProps); err != nil {
+		errs.Add(err)
+	}
+
+	if err := IngestRelationships(batch, ad.Entity, converted.RelProps); err != nil {
+		errs.Add(err)
+	}
+
+	return errs.Combined()
 }
 
-func IngestGroupData(batch graph.Batch, converted ConvertedGroupData) {
-	IngestNodes(batch, ad.Entity, converted.NodeProps)
-	IngestRelationships(batch, ad.Entity, converted.RelProps)
-	IngestDNRelationships(batch, converted.DistinguishedNameProps)
+func IngestGroupData(batch graph.Batch, converted ConvertedGroupData) error {
+	errs := util.NewErrorCollector()
+
+	if err := IngestNodes(batch, ad.Entity, converted.NodeProps); err != nil {
+		errs.Add(err)
+	} else if err = IngestRelationships(batch, ad.Entity, converted.RelProps); err != nil {
+		errs.Add(err)
+	} else if err = IngestDNRelationships(batch, converted.DistinguishedNameProps); err != nil {
+		errs.Add(err)
+	}
```

Comment:
I see you fixed one of the chained if blocks, but there's multiple others still 🙂